packages/outline-view/src/browser/outline-view-tree-model.ts

```diff
@@ -0,0 +1,55 @@
+/*
+* Copyright (C) 2017 TypeFox and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+import { injectable, inject } from "inversify";
+import { OutlineViewTree, SymbolInformationNode } from './outline-view-tree';
+import { TreeModel, ITreeNode, OpenerOptions, open, TreeServices, OpenerService } from "@theia/core/lib/browser";
+import { EditorManager } from "@theia/editor/lib/browser";
+import { Range } from '@theia/editor/lib/browser';
+
+@injectable()
+export class OutlineViewTreeServices extends TreeServices {
+    @inject(OpenerService) readonly openerService: OpenerService;
+}
+
+@injectable()
+export class OutlineViewTreeModel extends TreeModel {
+
+    protected readonly openerService: OpenerService;
+
+    constructor(
+        @inject(OutlineViewTree) protected readonly outlineViewTree: OutlineViewTree,
+        @inject(OutlineViewTreeServices) protected readonly outlineViewTreeServices: OutlineViewTreeServices,
+        @inject(EditorManager) protected readonly editorManager: EditorManager
+    ) {
+        super(outlineViewTree, outlineViewTreeServices);
+
+        this.selection.onSelectionChanged(node => {
```

Comment:
this needs to go to the monaco extension and hide behind an `onSelect` callback.